### PR TITLE
Clean and docs

### PR DESCRIPTION
**Pull request by ddgenome**
Clean up TypeScript code, including removing as many tslint warnings as possible.

Document things in README.

### CODE REVIEW

**Comment by ddgenome** [comment]
I kept this as two different commits since people may not want the cleanup.

File: README.md [hunk 1]
```diff
@@ -1,18 +1,77 @@
...
-npm install -G typescript
-npm test
+[node]: https://nodejs.org/en/
+[rugs]: https://www.npmjs.com/package/@atomist/rugs
+[ts]: https://www.typescriptlang.org/
+[docs]: http://docs.atomist.com/
 
-## Releasing
+This module contains some helper classes and interfaces for writing
+Rugs, but its main purpose is to bring in all the dependencies you
```
Comment (kipz) [inline README.md]: I wouldn't say that bringing in dependencies is its main purpose...but it _is_ handy.

File: util/tree/QueryByExample.ts [hunk 2]
```diff
@@ -20,9 +36,9 @@ export function match(a) {
  * @type L type of leaf (may be the same)
  */
 export function byExample<R extends GraphNode, L extends GraphNode>(g: any): PathExpression<R, L> {
-    let pathExpression = `/${queryByExampleString(g).path}`
-    console.log(`Created path expression [${pathExpression}] for ${JSON.stringify(g)}`)
-    return new PathExpression<R, L>(pathExpression)
+    const pathExpression = `/${queryByExampleString(g).path}`;
+    // console.log(`Created path expression [${pathExpression}] for ${JSON.stringify(g)}`);
```
Comment (kipz) [inline util/tree/QueryByExample.ts]: We should remove this entirely IMO

File: util/tree/QueryByExample.ts [hunk 3]
```diff
@@ -46,148 +62,144 @@ class Branch {
 // Internal state of query string generation
 class PathBuilderState {
 
-    private isMatch: boolean
-    private simplePredicates = ""
-    private complexPredicates = ""
-    private rootExpression: string
+    private isMatch: boolean;
+    private simplePredicates = "";
```
Comment (kipz) [inline util/tree/QueryByExample.ts:65]: What's with these default values of ""?

File: util/tree/QueryByExample.ts [hunk 4]
```diff
@@ -46,148 +62,144 @@ class Branch {
...
+        handlePrimitive(state, id, value);
+    } else {
+        // console.log(`Don't know what to do with unfamiliar result of invoking [${id}] was [${value}]`);
     }
 }
 
 function handlePrimitive(state: PathBuilderState, id: string, value) {
-    //console.log(`Non graph node result of invoking ${id} was [${value}]`)
-    state.addSimplePredicate(`[@${id}='${value}']`)
+    // console.log(`Non graph node result of invoking ${id} was [${value}]`);
```
Comment (kipz) [inline util/tree/QueryByExample.ts]: remove entirely IMO

**Review by kipz — APPROVED** [review]

**Comment by kipz** [comment]
LGTM

File: README.md (hunk 1 above)
Comment (kipz) [reply README.md]: The main purpose of this module is to provide a place for us to delivery TS/JS stuff on an independent release schedule from Rug itself.

File: util/tree/QueryByExample.ts (hunk 3 above)
Comment (ddgenome) [reply util/tree/QueryByExample.ts:65]: I was just fixing tslint issues, in this case adding semicolons, so I'm not sure what the intent here is, if any.

**Comment by ddgenome** [comment]
Thanks, feedback incorporated.

**Pull request merged**